### ISSUE TITLE
LIBERO Reproducibility

### ISSUE THREAD

**Issue by zhang2514yuchi** [openvla/openvla#223]
I am currently working on reproducing results for OpenVLA on the LIBERO Spatial task. I have followed training procedures that are nearly identical to those described in the provided materials. However, I have encountered an issue where the success rate remains at zero consistently.
To rule out potential training discrepancies, I switched to the official pretrained checkpoint, but the results remain unchanged. Given this, I suspect that the issue may be related to the environment setup, particularly the Docker image or dependencies.
Would it be possible to provide guidance on the recommended approach for setting up the exact environment used in your experiments? Specifically, do you have a Docker image or a list of required dependencies that could help ensure a faithful reproduction of the results?

absl-py                       2.1.0
accelerate                    1.4.0
annotated-types               0.7.0
antlr4-python3-runtime        4.9.3
array_record                  0.6.0
astunparse                    1.6.3
attrs                         25.1.0
bddl                          1.0.1
beautifulsoup4                4.13.3
bitsandbytes                  0.45.3
cachetools                    5.5.1
certifi                       2025.1.31
charset-normalizer            3.4.1
click                         8.1.8
cloudpickle                   2.1.0
contourpy                     1.3.1
cycler                        0.12.1
debugpy                       1.8.12
dlimp                         0.0.1
dm-tree                       0.1.9
docker-pycreds                0.4.0
draccus                       0.10.0
easydict                      1.9
egl-probe                     1.0.2
einops                        0.4.1
etils                         1.12.0
evdev                         1.9.0
exceptiongroup                1.2.2
Farama-Notifications          0.0.4
fastjsonschema                2.21.1
filelock                      3.17.0
flatbuffers                   25.2.10
fonttools                     4.56.0
fsspec                        2025.2.0
future                        0.18.2
gast                          0.6.0
gdown                         5.2.0
gitdb                         4.0.12
GitPython                     3.1.44
glfw                          2.8.0
google-auth                   2.38.0
google-auth-oauthlib          1.2.1
google-pasta                  0.2.0
grpcio                        1.70.0
gym                           0.25.2
gym-notices                   0.0.8
gymnasium                     0.29.1
h5py                          3.13.0
huggingface-hub               0.29.1
hydra-core                    1.2.0
idna                          3.10
imageio                       2.37.0
imageio-ffmpeg                0.6.0
importlib_resources           6.5.2
iniconfig                     2.0.0
Jinja2                        3.1.5
joblib                        1.4.2
json-numpy                    2.1.0
jsonlines                     4.0.0
jsonschema                    4.23.0
jsonschema-specifications     2024.10.1
jupyter_core                  5.7.2
jupytext                      1.16.7
keras                         3.8.0
kiwisolver                    1.4.8
libclang                      18.1.1
libero                        0.1.0                /root/yczhang/LIBERO
llvmlite                      0.43.0
mani_skill2_real2sim          0.5.3                /root/yczhang/SimplerEnv/ManiSkill2_real2sim
Markdown                      3.7
markdown-it-py                3.0.0
MarkupSafe                    3.0.2
matplotlib                    3.5.3
mdit-py-plugins               0.4.2
mdurl                         0.1.2
mergedeep                     1.3.4
ml-dtypes                     0.4.1
mpmath                        1.3.0
mujoco                        3.2.4
MultiScaleDeformableAttention 1.0
mypy-extensions               1.0.0
namex                         0.0.8
nbformat                      5.10.4
networkx                      3.4.2
ninja                         1.11.1.3
nltk                          3.9.1
numba                         0.60.0
numpy                         1.26.0
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu12             8.9.2.26
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu12          12.1.0.106
nvidia-nccl-cu12              2.19.3
nvidia-nvjitlink-cu12         12.8.61
nvidia-nvtx-cu12              12.1.105
oauthlib                      3.2.2
omegaconf                     2.3.0
opencv-python                 4.6.0.66
OpenEXR                       3.3.2
openvla                       0.0.3                /root/yczhang/openvla
opt_einsum                    3.4.0
optree                        0.14.1
packaging                     24.2
pathtools                     0.1.2
peft                          0.11.1
pillow                        11.1.0
pip                           25.0.1
platformdirs                  4.3.6
pluggy                        1.5.0
promise                       2.3
protobuf                      3.20.3
psutil                        7.0.0
pyasn1                        0.6.1
pyasn1_modules                0.4.1
pydantic                      2.10.6
pydantic_core                 2.27.2
Pygments                      2.19.1
pynput                        1.7.7
PyOpenGL                      3.1.9
pyparsing                     3.2.1
PySocks                       1.7.1
pytest                        8.3.4
python-dateutil               2.9.0.post0
python-xlib                   0.33
PyYAML                        6.0.2
pyyaml-include                1.4.1
referencing                   0.36.2
regex                         2024.11.6
requests                      2.32.3
requests-oauthlib             2.0.0
rich                          13.9.4
robomimic                     0.2.0
robosuite                     1.4.0
rpds-py                       0.22.3
rsa                           4.9
Rtree                         1.3.0
ruckig                        0.14.0
safetensors                   0.5.2
sapien                        2.2.2
scipy                         1.15.2
sentencepiece                 0.1.99
sentry-sdk                    2.22.0
setproctitle                  1.3.4
setuptools                    59.6.0
simpler_env                   0.0.1                /root/yczhang/SimplerEnv
six                           1.17.0
smmap                         5.0.2
soupsieve                     2.6
sympy                         1.13.3
tabulate                      0.9.0
tensorboard                   2.18.0
tensorboard-data-server       0.7.2
tensorboardX                  2.6.2.2
tensorflow                    2.18.0
tensorflow-addons             0.23.0
tensorflow-datasets           4.9.3
tensorflow-estimator          2.15.0
tensorflow-graphics           2021.12.3
tensorflow-io-gcs-filesystem  0.37.1
tensorflow-metadata           1.16.1
termcolor                     2.5.0
thop                          0.1.1.post2209072238
timm                          0.9.10
tokenizers                    0.19.1
toml                          0.10.2
tomli                         2.2.1
torch                         2.2.0
torchaudio                    2.2.0
torchvision                   0.17.0
tqdm                          4.67.1
traitlets                     5.14.3
transformers                  4.40.1
transforms3d                  0.4.2
trimesh                       4.6.2
triton                        2.2.0
typeguard                     2.13.3
typing_extensions             4.12.2
typing-inspect                0.9.0
urllib3                       2.3.0
wandb                         0.19.6
Werkzeug                      3.1.3
wheel                         0.37.1
wrapt                         1.14.1
zipp                          3.21.0

**Comment by zhang2514yuchi** [comment]
We tested the model openvla/openvla-7b-finetuned-libero-spatial on huggingface, and the success rate was below 5% after multiple attempts.

I carefully compared the difference between the training data and the images rendered during our test, and found that our rendering had more aliasing and noise, but there was no other visible difference.

We tried EGL and osmesa, both of which had slight differences from the training images.

OpenGL version is 4.6
Our mujoco version is 3.2.2

**Comment by zhang2514yuchi** [comment]
I found the problem. I was using the code of openvla-mini. After careful inspection, I found that it had made very subtle changes internally, which caused serious test problems.
Sorry to bother you
